### PR TITLE
Added a search bar, only searching Users for now

### PR DESCRIPTION
**Pull request by lealgamb**
Added a search bar to the navigation bar. Empty searches are disabled. Currently only searching for users and getting/displaying them as a list. Will add tests in another PR.

Search Bar:
image

Empty Search:
image

Valid Search:
image

### CODE REVIEW

**Comment by jasminechau2** [comment]
It LGTM!

File: src/main/webapp/css/main.css [hunk 1]
```diff
@@ -35,6 +35,13 @@ nav a {
   text-decoration: none;
 }
 
+nav form {
```
Comment (vocatwk) [inline src/main/webapp/css/main.css]: Sorry couldn't find where you used this.

**Comment by vocatwk** [comment]
Not sure what you guys prefer but could we also have the search results displayed on top of a page as like a drop down instead of a different page?

File: src/main/webapp/WEB-INF/view/search.jsp [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        <p>No results for: <%= searchRequest %></p>
+      <%
+      }
+      else {
+      %>
+        <p>Results for: <%= searchRequest %></p>
+      <%
+        for (User user : users) {
+      %>
+          <p><%= user.getName() %></p>
```
Comment (yebrahim) [inline src/main/webapp/WEB-INF/view/search.jsp]: How about we make this a link to the user's profile page?

File: src/main/java/codeu/model/store/basic/UserStore.java [hunk 3]
```diff
@@ -98,6 +98,27 @@ public User getUser(UUID id) {
...
   }
 
+  /**
+  * Access the Users that contains the given username.
+  * 
+  * Used with search requests. 
+  * 
+  * @return null if username does not match any existing user
+  */
+  public List<User> getUsers(String username) {
```
Comment (yebrahim) [inline src/main/java/codeu/model/store/basic/UserStore.java]: Tests for this method?

File: src/main/java/codeu/controller/SearchServlet.java [hunk 4]
```diff
@@ -0,0 +1,91 @@
+package codeu.controller;
```
Comment (yebrahim) [inline src/main/java/codeu/controller/SearchServlet.java]: Tests for this servlet?

**Comment by yebrahim** [comment]
Nice work on the UI too, the search box looks good.

File: src/main/webapp/css/main.css (hunk 1 above)
Comment (lealgamb) [reply src/main/webapp/css/main.css]: It's for the form that the search bar is contained in.

File: src/main/webapp/WEB-INF/view/search.jsp (hunk 2 above)
Comment (lealgamb) [reply src/main/webapp/WEB-INF/view/search.jsp]: Will do, forgot to mention in the description that this was only to test the search function.

File: src/main/java/codeu/controller/SearchServlet.java (hunk 4 above)
Comment (lealgamb) [reply src/main/java/codeu/controller/SearchServlet.java]: Will add the tests in a future PR, sorry I forgot to mention that.

**Comment by lealgamb** [comment]
@vocatwk I will add that feature either in this PR or a new one. Are you thinking of a dynamic drop down menu, like it searches as you type? And if the user clicks the button then it redirects to the other page?

**Comment by lealgamb** [comment]
I'm having some trouble figuring out how to implement the instant search. Any tips?

**Comment by vocatwk** [comment]
@lealgamb that sounds good to me!

**Comment by vocatwk** [comment]
is there a way to detect if the value of search bar changes and update your search results each time it does. and you could preset the value to the value you just used to search for data so that you don't interrupt the user's search. It sounds very expensive but can't think about any other way of doing it at the moment.

**Comment by yebrahim** [comment]
Looks good, but can we merge the PR if it’s ready with the separate results page, then work on the dynamic drop down in a separate PR?

**Comment by yebrahim** [comment]
After merging this, can we add a little section in the PR about search? Just a sentence or two about the fact that we offer searching for usernames right from the top bar.

**Comment by lealgamb** [comment]
Yup, will merge this now.

**Pull request merged**